internal/server/v1beta1/search.go

```diff
@@ -49,6 +50,61 @@ func (server *APIServer) SearchAssets(ctx context.Context, req *compassv1beta1.S
 	}, nil
 }
 
+func (server *APIServer) GroupAssets(ctx context.Context, req *compassv1beta1.GroupAssetsRequest) (*compassv1beta1.GroupAssetsResponse, error) {
+	_, err := server.validateUserInCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	groupby := req.GetGroupby()
+	if groupby == "" {
+		return nil, status.Error(codes.InvalidArgument, "'groupby' must be specified")
+	}
+
+	cfg := asset.GroupConfig{
+		GroupBy: strings.Split(groupby, ","),
+		Filters: filterConfigFromValues(req.GetFilter()),
+		Size:    int(req.GetSize()),
+	}
+
+	includedFields := req.GetIncludeFields()
+	if includedFields != "" {
+		cfg.IncludedFields = strings.Split(includedFields, ",")
+	}
+
+	results, err := server.assetService.GroupAssets(ctx, cfg)
+	if err != nil {
+		return nil, internalServerError(server.logger, fmt.Sprintf("error searching asset: %s", err.Error()))
+	}
+
+	groupInfoArr := make([]*compassv1beta1.GroupAssetInfo, len(results))
+	for idx, gr := range results {
+		assetsPB := make([]*compassv1beta1.Asset, len(gr.Assets))
+		for assetIdx, as := range gr.Assets {
+			assetPB, err := assetToProto(as, false)
+			if err != nil {
+				return nil, internalServerError(server.logger, fmt.Sprintf("error converting assets to proto: %s", err.Error()))
+			}
+			assetsPB[assetIdx] = assetPB
+		}
+
+		groupInfo := &compassv1beta1.GroupAssetInfo{
+			GroupFieldInfo: []*compassv1beta1.GroupFieldInfo{
+				{
+					GroupKey:   cfg.GroupBy[0],
```

Comment:
only for the first key ?

Reply:
yes, currently we have added support for grouping on only 1 field.

Reply:
This has been fixed